### PR TITLE
finalize overview and exercise section for tasks

### PR DESCRIPTION
**Pull request by tkellen**

### CODE REVIEW

File: workshop/exercises/tasks/overview.md [hunk 1]
```diff
@@ -17,15 +17,53 @@ git using the apt package manager.
...
+  - apt: name=git state=latest update_cache=true
+```
+
+Help files are available for all modules via the `ansible-doc` command. To see
+what the `update_cache` option does for the `apt` module, run `ansible-doc apt`.
+
+Documentation for all modules is also available on the web at:
+http://docs.ansible.com.
+
+## NAMING TASKS
```
Comment (MattSurabian) [inline workshop/exercises/tasks/overview.md]: We should mention how the names are used in the output while the command is running

File: workshop/exercises/tasks/overview.md [hunk 2]
```diff
@@ -17,15 +17,53 @@ git using the apt package manager.
...
-for "debug" at http://docs.ansible.com.
+## IDEMPOTENCY
+
+Most Ansible tasks are idempotent by default. This means that you can run the
+same task repeatedly and your server will remain in the same state. This feature
+may not seem compelling when considering something as simple as installing a
+piece of software, but it quickly becomes critical as your tasks get more
+complex.
+
+For example, imagine running a task to add a line to a configuration file. Even
```
Comment (MattSurabian) [inline workshop/exercises/tasks/overview.md]: clutch example

File: workshop/exercises/tasks/overview.md [hunk 3]
```diff
@@ -17,15 +17,53 @@ git using the apt package manager.
...
+if you run that task twenty times, you only want the line added once (like most
+things, Ansible has a module for that, it's called `lineinfile`).
+
+## EXERCISE
+
+Try creating a playbook that installs the apt packages `git`, `nodejs` and
+`nginx`. Try running this playbook with and without names for the tasks to
+see how the output changes.
```
Comment (MattSurabian) [inline workshop/exercises/tasks/overview.md]: Ah! My comment above can be ignored.

**Comment by MattSurabian** [comment]
:+1:

File: workshop/exercises/tasks/overview.md (hunk 2 above)
Comment (tkellen) [inline workshop/exercises/tasks/overview.md]: :)

**Comment by tkellen** [comment]
woo

**Pull request merged**